app/controllers/application_controller.rb

```diff
@@ -47,10 +47,19 @@ def check_your_privilege(name, post = nil, render_error = true)
   private
 
   def set_globals
+    RequestContext.clear!
+
+    host_name = request.raw_host_with_port   # include port to support multiple localhost instances
+    RequestContext.community = @community = Rails.cache.fetch("#{host_name}/community", expires_in: 30.minutes) do
```

Comment:
Could probably make this non-expiring - if the community for a particular host is going to change, it wouldn't be unreasonable to require a server restart

Reply:
Are we assuming memory cache is being used here?  If using file or Redis, it won't be removed by a restart.  I'll remove either way, just raising awareness.

Reply:
Aye, point - I was assuming memory cache, yes. What if we change it based on cache strategy in use, i.e. (pseudocode) `expires_in: Rails.cache.strategy.memory? ? 365.days : 1.hour`?

Reply:
That could work.  My personal opinion is that having a straight `1.hour` timeout should suffice.  Reasoning being that a single DB lookup per community once an hour is not going to make any discernible difference, and this approach keeps the code simple.